lib/c-api/src/wasm_c_api/module.rs

```diff
@@ -115,3 +115,364 @@ pub unsafe extern "C" fn wasm_module_serialize(
     };
     *out_ptr = byte_vec.into();
 }
+
+#[cfg(test)]
+mod tests {
+    use inline_c::assert_c;
+
+    #[test]
+    fn test_module_validate() {
+        (assert_c! {
+            #include "tests/wasmer_wasm.h"
+
+            int main() {
+                wasm_engine_t* engine = wasm_engine_new();
+                wasm_store_t* store = wasm_store_new(engine);
+
+                wasm_byte_vec_t wat;
+                wasm_byte_vec_new_from_string(&wat, "(module)");
+                wasm_byte_vec_t* wasm = wat2wasm(&wat);
+
+                assert(wasm_module_validate(store, wasm));
```

Comment:
Unlike `assert!` in rust, this does not run `wasm_module_validate` in a release build. The assert() macro is defined to nothing in release builds. Therefore this won't test module_validate.

Reply:
Hmm, so we'll have to write our own `assert` :-).

Reply:
Our own `wasmer_assert` macro has been implemented. It mimics the standard `assert` macro, with a different name, to avoid confusion.